tests/lib/devicetree/src/main.c

```diff
@@ -881,6 +881,76 @@ static void test_io_channels(void)
 	zassert_equal(DT_INST_IO_CHANNELS_INPUT(0), 10, "input 0 implicit");
 }
 
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_spi
+static void test_dma(void)
+{
+	zassert_true(!strcmp(DT_LABEL(DT_DRV_INST(0)), "TEST_SPI_CTLR"),
```

Comment:
ok, then, since I just reused "test_spi" node, do I really need to add an additional test? Or is it just superfluous?

Reply:
Sorry I was vague. All of the new inst macros need test cases. However, the TEST_SPI node is not the right place to add the "dmas" property which is used to test these macros.

That is because there are two instances of "vnd,spi" in the devicetree. This means that hard-coding instance number 0, for example as done here:

```
	zassert_equal(DT_INST_DMAS_CELLS_BY_NAME(0, rx, channel), 3,
		      "channel cell of rx dma channel");
```

Is not correct, because there is no guarantee that `/test/spi@33334444` is instance number 0 of "vnd,spi". It could be `/test/spi@55556666`; the API does not make a guarantee.

The solution is to put the "dmas" and "dma-names" properties inside of a node whose compatible has exactly one instance in the entire devicetree. Then the API guarantee that the instances are numbered starting at 0 for every enabled node means that the above test would be safe.

So I am suggesting doing something like this instead:

```
		/* there should only be one of these */
		test_temp_sensor: temperature-sensor {
			compatible = "vnd,adc-temp-sensor";
			label = "TEST_TEMP";
			dmas = <&test_dma1 1 2>, <&test_dma2 3 4>;
			dma-names = "tx", "rx";
			io-channels = <&test_adc_1 10>, <&test_adc_2 20>;
			io-channel-names = "ch1", "ch2";
			clocks = <&test_clk 3 7>, <&test_fixed_clk>, <&test_clk 8 2>;
		};
```

And doing the tests with DT_DRV_COMPAT set to vnd_adc_temp_sensor.

You can pick any node that we ensure there's only one of. This is just a suggestion.

Reply:
> Is not correct, because there is no guarantee that /test/spi@33334444 is instance number 0 of "vnd,spi". It could be /test/spi@55556666; the API does not make a guarantee.

Ok. This was indeed what I was trying to check.
I get your point now, I'll take your suggestion.

Reply:
@mbolivar-nordic  Done